### PR TITLE
feat(donations): add unauthenticated fiat donation flow

### PR DESCRIPTION
**Pull request by brunodmsi**
## Overview

Enable fiat donations without requiring wallet login. Unauthenticated users provide their email, complete a Stripe payment, and track donation status via polling tokens — no wallet connection needed.

**Backend PR**: https://github.com/show-karma/gap-indexer/pull/893

## Problem

The fiat donation flow previously required users to connect a wallet before accessing the payment tab. This creates unnecessary friction — many users willing to donate via credit card don't have or want to connect a crypto wallet.

### Unauthenticated OnrampFlow
- New `isAuthenticated` prop on `OnrampFlow` — when `false`, shows an email input field
- Email validated with strict regex that rejects edge cases (consecutive dots, leading hyphens, etc.)
- `emailTouched` state with `onBlur` validation to avoid premature error display
- Specific error messages for empty vs. invalid email states
- Email trimmed with `.trim()` before sending to prevent whitespace-only submissions
- `SingleProjectDonateModal` removes the login gate for the FIAT tab
- Email passed to backend as `donorEmail` for Stripe receipt and donation tracking

### Polling Token Status Tracking
- `useDonationPolling` hook supports a `pollingToken` parameter
- When present, calls `getDonationStatus()` (public, no auth) instead of `getDonationByUid()` (authenticated)
- Polling token sent via `X-Polling-Token` header (not query param) to prevent token leaking in server logs and browser history
- Polling token excluded from React Query cache key to prevent credential exposure in DevTools

### Authenticated Donation History
- New `GET /v2/donations/me` endpoint used via `useMyDonations` hook
- Merges wallet-based and email-based donation history for authenticated users
- Removed dead `getUserDonations`/`getProjectDonations` service methods and their query keys

### Type Safety
- `DonationStatusApiResponse.status` typed as `DonationStatus` enum instead of inline string union
- `toDonationStatus()` type guard validates status values against `Object.values(DonationStatus)` set, replacing unsafe `as DonationStatus` cast
- `donorAddress` made optional (`donorAddress?: string`) in `DonationApiResponse` and `OnrampSessionRequest`
- `pollingToken` type aligned to accept `string | null` across components and hooks

### Accessibility (WCAG 2.2 AA)
- `aria-invalid` on email input when validation error is present
- `aria-describedby="email-error"` linking input to its error message
- Error `<p>` element with `role="alert"` and `id="email-error"` for screen reader announcements
- Matches the existing amount field accessibility pattern

### Data Flow
- `OnrampFlow` captures `pollingToken` from session via ref (same pattern as `donationUid`)
- Passes through `OnrampSuccessModal` → `useDonationPolling`
- Modal shows real-time status updates regardless of auth state

### Security

| Issue | Fix |
|-------|-----|
| **Polling token in React Query cache key** (P1) | Removed `pollingToken` from `QUERY_KEYS.DONATIONS.STATUS` cache key. Token is still passed to `queryFn` but no longer visible in DevTools. |

### Code Quality

| Issue | Fix |
|-------|-----|
| **Email regex allowed invalid edge cases** (P2) | Replaced with stricter `EMAIL_REGEX` constant — rejects `user@.com`, `user@-domain.com`, `user@domain..com`, `user..name@domain.com` |
| **`console.error` in production interceptors** (P3) | Removed both `apiClient` and `publicApiClient` error interceptors — they only logged and re-threw; errors already handled by React Query `onError` callbacks |
| **`pollingToken` type mismatch** (P3) | Aligned `UseDonationPollingParams.pollingToken` to accept `string \| null`, removed `?? undefined` coercion in `OnrampSuccessModal` |
| **Dead service methods and query keys** | Removed unused `getUserDonations`, `getProjectDonations`, `BY_USER`, `BY_PROJECT` query keys, and no-op cache invalidation in `useCreateDonation` |

## Why This Approach

- **Email as identifier**: Stripe requires customer email for receipts. Collecting it also enables donation tracking notifications without requiring account creation.
- **Polling token over session auth**: The polling token is scoped to a single donation and validated server-side. It enables status tracking without JWT while keeping the endpoint secure.
- **Header-based token transport**: `X-Polling-Token` header avoids token exposure in server access logs, proxy logs, and browser history — unlike query parameters.
- **Token excluded from cache key**: The polling token is a security credential; storing it only in the `queryFn` closure minimizes its exposure surface.
- **Type guard over type assertion**: `toDonationStatus()` validates against `Object.values(DonationStatus)` at runtime, returning `null` for unexpected values instead of silently accepting any string.
- **Conditional UI, shared components**: Same `OnrampFlow` and `OnrampSuccessModal` components serve both authenticated and anonymous users — only the email input and API path differ.

## Architecture

```mermaid
sequenceDiagram
    participant U as User (no wallet)
    participant M as SingleProjectDonateModal
    participant F as OnrampFlow
    participant H as useOnramp
    participant API as Backend
    participant S as StripeOnrampEmbed
    participant SM as OnrampSuccessModal
    participant P as useDonationPolling

    U->>M: Click "Pay with Card"
    M->>F: isAuthenticated=false
    F->>F: Show email input
    U->>F: Enter email + amount
    F->>H: initiateOnramp(amount, currency)
    H->>API: POST /v2/onramp/session { email, amount }
    API-->>H: { sessionToken, donationUid, pollingToken }
    H-->>F: session with pollingToken
    F->>F: Capture donationUid + pollingToken in refs
    F->>S: clientSecret
    U->>S: Complete Stripe payment
    S-->>F: onSuccess(sessionData)
    F->>F: clearSession(), show success modal
    F->>SM: donationUid, pollingToken, chainId
    SM->>P: useDonationPolling({ donationUid, chainId, pollingToken })
    P->>API: GET /v2/donations/:uid/:chainId/status (X-Polling-Token header)
    API-->>P: { status, amount, txHash }
    P-->>SM: Real-time status updates
```

## Testing

All **144 donation-related tests** pass across 5 test suites. Full project: **308 suites, 6559 tests passing**.

| Suite | Covers |
|-------|--------|
| `OnrampFlow.test.tsx` | Email input validation (edge cases: `@.com`, `@-domain`, `..com`), auth states, amount validation, accessibility |
| `OnrampSuccessModal.test.tsx` | Polling token path, type alignment |
| `useDonationPolling.test.tsx` | Authenticated vs. anonymous polling, terminal states, cache key security (no token in key) |
| `useOnramp.test.tsx` | `donorEmail` and `pollingToken` handling |
| `queryKeys.test.ts` | Key factory consistency, no credential leakage |

## Testing Steps

1. **Unauthenticated flow**: Without logging in, open a project -> Donate -> Pay with Card tab
   - Email input should be visible
   - Click away from email input (blur) with empty field -> "Email is required" error
   - Enter invalid email -> "Please enter a valid email address" error
   - Button disabled until both valid email and valid amount entered
   - After Stripe payment, success modal polls status via `X-Polling-Token` header
2. **Authenticated flow**: Log in -> same flow, but no email input shown; uses `/me` endpoint for history
3. **Email trimming**: Enter email with trailing spaces -> should be trimmed before submission
4. **Amount validation**: Min $10, Max $10,000, max 2 decimal places
5. **Accessibility**: Tab through email input, verify screen reader announces errors via `aria-invalid` and `role="alert"`
6. **Terminal state handling**: Polling stops automatically when status reaches `completed` or `failed`

## Files Changed

| File | Change |
|------|--------|
| `components/Donation/SingleProject/OnrampFlow.tsx` | `isAuthenticated` prop, email input with strict `EMAIL_REGEX` validation/a11y, `pollingTokenRef`, `emailTouched` state |
| `components/Donation/SingleProject/OnrampSuccessModal.tsx` | Pass `pollingToken` to polling hook, simplified type coercion |
| `components/Donation/SingleProject/SingleProjectDonateModal.tsx` | Remove login gate for FIAT tab |
| `hooks/donation/useDonationPolling.ts` | Support `pollingToken` path (token in `queryFn` only, not cache key), `toDonationStatus()` type guard, `pollingToken?: string \| null` type |
| `hooks/donation/useOnramp.ts` | Accept optional `donorEmail` param, trim before sending |
| `hooks/donation/useCreateDonation.ts` | Simplified to only invalidate `MY()` query key |
| `hooks/donation/types.ts` | `DonationStatus` enum, `DonationStatusApiResponse`, optional `donorAddress`/`donorEmail`, `pollingToken` in session |
| `services/donations.service.ts` | `publicApiClient`, `getDonationStatus()` with `X-Polling-Token` header, removed redundant error interceptors |
| `utilities/queryKeys.ts` | `QUERY_KEYS.DONATIONS.STATUS()` without token param, removed dead `BY_USER`/`BY_PROJECT` keys |
| `components/Donation/SingleProject/__tests__/OnrampFlow.test.tsx` | Unauthenticated flow tests, email edge case validation |
| `components/Donation/SingleProject/__tests__/OnrampSuccessModal.test.tsx` | Polling token path tests |
| `hooks/donation/__tests__/useDonationPolling.test.tsx` | Full polling hook test suite, cache key security test |
| `hooks/donation/__tests__/useOnramp.test.tsx` | Updated for `donorEmail` and `pollingToken` |

## Checklist

- [x] Tests added for unauthenticated flow (email input, validation, button states)
- [x] Tests added for polling token path in `useDonationPolling`
- [x] Tests verify polling token is NOT in React Query cache key
- [x] Tests cover email regex edge cases (consecutive dots, leading hyphens)
- [x] `DonationStatus` enum used throughout instead of string literals
- [x] `toDonationStatus()` type guard replaces unsafe cast
- [x] Polling token sent via `X-Polling-Token` header (not query param)
- [x] Polling token excluded from cache key (security)
- [x] Email validated with strict regex, trimmed before submission
- [x] Uses centralized `QUERY_KEYS` for cache management
- [x] Dead service methods and query keys removed
- [x] Redundant error interceptors removed
- [x] `pollingToken` type aligned across components and hooks
- [x] Accessibility: `aria-invalid`, `aria-describedby`, `role="alert"` on email error
- [x] No breaking changes to authenticated donation flow
- [x] 308 test suites, 6559 tests passing

### CODE REVIEW

**Review by brunodmsi — COMMENTED** [review]
## PR Review: feat(donations): add unauthenticated fiat donation flow

**14 files** | +334 / -55 | Branch: `feat/fiat-unauth` → `main`

### Summary

Solid, well-scoped feature that removes the wallet-login gate for fiat donations. The architecture is clean — conditional auth paths share components, polling token enables anonymous status tracking, and test coverage is thorough. A few issues worth addressing before merge.

---

### Critical Issues (P0/P1)

**1. [P1] Email regex is too permissive — allows injection-friendly inputs**
`OnrampFlow.tsx` ~L84:
```ts
return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(donorEmail);
```
This passes strings like `"><script>@x.y`, `user@.com`, or single-char TLDs. Since this email goes to the backend and eventually to Stripe, the validation should be tighter.

**Suggested fix** — use a stricter pattern:
```ts
return /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(donorEmail);
```
Or use the native `input type="email"` validity API via a ref.

**2. [P1] `donorEmail` is required in `OnrampSessionRequest` but sent empty for authenticated users**
`types.ts`: `donorEmail: string` is required (not optional). For authenticated users, the email field is hidden and `donorEmail` stays `""`, which gets sent to the backend.

**Suggested fix** in `useOnramp.ts`:
```ts
...(donorEmail && { donorEmail }),
```
And in `types.ts`:
```ts
donorEmail?: string;
```

**3. [P1] `isAuthenticated` prop defaults to `undefined` — email shows unexpectedly**
If any caller forgets to pass `isAuthenticated`, it defaults to `undefined` (falsy), so the email field shows up and blocks submission. This is a subtle breaking change for any other consumers of `OnrampFlow`.

**Suggested fix**: Default to `true` to preserve backward compatibility:
```ts
({ projectUid, payoutAddress, chainId, initialAmount, isAuthenticated = true, onDonationComplete })
```

---

### Other Issues (P2/P3)

**4. [P2] Polling token leaked in URL query string**
`donations.service.ts`: `{ params: { pollingToken } }` — token ends up in server access logs, browser history, HTTP logging. Consider sending as a header (`X-Polling-Token`) if it grants data access.

**5. [P2] Union return type weakens type safety**
`useDonationPolling.ts`: `donation: DonationApiResponse | DonationStatusApiResponse | null` — callers need to check which shape they got. Consider a discriminated union or generic.

**6. [P2] Unsafe cast on status**
`useDonationPolling.ts`: `(data?.status as DonationStatus)` — if enum values drift from API response strings, bugs appear silently. Consider explicit mapping.

**7. [P3] Unrelated changes inflate the diff**
- `.claude/settings.local.json` reformatting
- `programRegistry.service.test.ts` removed `anyoneCanJoin` (unrelated)
- JSX reformatting in `OnrampSuccessModal.tsx`
- Import reordering in test file

**8. [P3] Missing `publicApiClient` error interceptor**
The authenticated `apiClient` has an error interceptor, but `publicApiClient` doesn't. Anonymous users will see raw Axios errors instead of normalized format.

---

### Suggestions

- Add an integration-level test covering the full flow: email input → `initiateOnramp` with `donorEmail` → session with `pollingToken` → success modal receives it.
- Consider debouncing email validation (minor perf improvement).
- `isAuthenticated` prop name is ambiguous — `requiresEmailInput` or `isAnonymous` (inverted) might communicate intent better.

---

### Score: 7/10

**Merge recommendation: Approve with changes.** Fix P1 issues #2 (empty `donorEmail` for authed users) and #3 (default `isAuthenticated` to `true`). Issue #1 (email regex) should be tightened before production. The rest are P2/P3 improvements that can be addressed in follow-up.

**Review by brunodmsi — COMMENTED** [review]
## PR Review: feat(donations): add unauthenticated fiat donation flow

**14 files** | +334 / -55 | Branch: `feat/fiat-unauth` → `main`

### Summary

Solid, well-scoped feature that removes the wallet-login gate for fiat donations. The architecture is clean — conditional auth paths share components, polling token enables anonymous status tracking, and test coverage is thorough. A few issues worth addressing before merge.

---

### Critical Issues (P0/P1)

**1. [P1] Email regex is too permissive — allows injection-friendly inputs**
`OnrampFlow.tsx` ~L84:
```ts
return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(donorEmail);
```
This passes strings like `"><script>@x.y`, `user@.com`, or single-char TLDs. Since this email goes to the backend and eventually to Stripe, the validation should be tighter.

**Suggested fix** — use a stricter pattern:
```ts
return /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(donorEmail);
```
Or use the native `input type="email"` validity API via a ref.

**2. [P1] `donorEmail` is required in `OnrampSessionRequest` but sent empty for authenticated users**
`types.ts`: `donorEmail: string` is required (not optional). For authenticated users, the email field is hidden and `donorEmail` stays `""`, which gets sent to the backend.

**Suggested fix** in `useOnramp.ts`:
```ts
...(donorEmail && { donorEmail }),
```
And in `types.ts`:
```ts
donorEmail?: string;
```

**3. [P1] `isAuthenticated` prop defaults to `undefined` — email shows unexpectedly**
If any caller forgets to pass `isAuthenticated`, it defaults to `undefined` (falsy), so the email field shows up and blocks submission. This is a subtle breaking change for any other consumers of `OnrampFlow`.

**Suggested fix**: Default to `true` to preserve backward compatibility:
```ts
({ projectUid, payoutAddress, chainId, initialAmount, isAuthenticated = true, onDonationComplete })
```

---

### Other Issues (P2/P3)

**4. [P2] Polling token leaked in URL query string**
`donations.service.ts`: `{ params: { pollingToken } }` — token ends up in server access logs, browser history, HTTP logging. Consider sending as a header (`X-Polling-Token`) if it grants data access.

**5. [P2] Union return type weakens type safety**
`useDonationPolling.ts`: `donation: DonationApiResponse | DonationStatusApiResponse | null` — callers need to check which shape they got. Consider a discriminated union or generic.

**6. [P2] Unsafe cast on status**
`useDonationPolling.ts`: `(data?.status as DonationStatus)` — if enum values drift from API response strings, bugs appear silently. Consider explicit mapping.

**7. [P3] Unrelated changes inflate the diff**
- `.claude/settings.local.json` reformatting
- `programRegistry.service.test.ts` removed `anyoneCanJoin` (unrelated)
- JSX reformatting in `OnrampSuccessModal.tsx`
- Import reordering in test file

**8. [P3] Missing `publicApiClient` error interceptor**
The authenticated `apiClient` has an error interceptor, but `publicApiClient` doesn't. Anonymous users will see raw Axios errors instead of normalized format.

---

### Suggestions

- Add an integration-level test covering the full flow: email input → `initiateOnramp` with `donorEmail` → session with `pollingToken` → success modal receives it.
- Consider debouncing email validation (minor perf improvement).
- `isAuthenticated` prop name is ambiguous — `requiresEmailInput` or `isAnonymous` (inverted) might communicate intent better.

---

### Score: 7/10

**Merge recommendation: Approve with changes.** Fix P1 issues #2 (empty `donorEmail` for authed users) and #3 (default `isAuthenticated` to `true`). Issue #1 (email regex) should be tightened before production. The rest are P2/P3 improvements that can be addressed in follow-up.

**Comment by brunodmsi** [comment]
## All flagged issues are already resolved at HEAD

The latest review appears to be analyzing a stale diff. All 5 items raised have been fixed in prior commits on `feat/fiat-unauth`. Here's the evidence:

### 1. Polling token in URL → Fixed: sent as header
**`services/donations.service.ts:77`** — token is sent via `X-Polling-Token` header, not as a query parameter:
```ts
{ headers: { "X-Polling-Token": pollingToken } }
```

### 2. Unsafe type assertion → Fixed: validated with `Set.has()` guard
**`hooks/donation/useDonationPolling.ts:14-17`** — `toDonationStatus()` validates the string against enum values before casting:
```ts
const DONATION_STATUS_VALUES = new Set<string>(Object.values(DonationStatus));
const toDonationStatus = (s?: string | null): DonationStatus | null =>
  s && DONATION_STATUS_VALUES.has(s) ? (s as DonationStatus) : null;
```

### 3. Type duplication (string literals vs enum) → Fixed: uses `DonationStatus` directly
**`hooks/donation/types.ts:72`** — `DonationStatusApiResponse.status` uses the enum, not string literals:
```ts
export interface DonationStatusApiResponse {
  status: DonationStatus;
  // ...
}
```

### 4. Email accessibility → Fixed: full a11y attributes present
**`components/Donation/SingleProject/OnrampFlow.tsx:205-214`** — email input has `aria-required`, `aria-invalid`, `aria-describedby`, and a visible error with `role="alert"`:
```ts
aria-describedby={emailTouched && emailError ? "donor-email-error" : "donor-email-hint"}
aria-required="true"
aria-invalid={emailTouched && emailError ? "true" : undefined}
// ...
<p id="donor-email-error" role="alert" className="text-xs text-red-500">
  {emailError}
</p>
```

### 5. Empty `donorEmail` for authenticated users → Fixed: conditionally spread
**`hooks/donation/useOnramp.ts:69`** — empty strings are excluded via conditional spread:
```ts
...(donorEmail?.trim() && { donorEmail: donorEmail.trim() }),
```

All items can be verified by inspecting the current HEAD of `feat/fiat-unauth`. No code changes needed.

File: components/Donation/SingleProject/OnrampFlow.tsx [hunk 1]
```diff
@@ -163,6 +175,29 @@ export const OnrampFlow = React.memo<OnrampFlowProps>(
...
+              value={donorEmail}
+              onChange={(e) => setDonorEmail(e.target.value)}
+              aria-describedby="donor-email-hint"
+              className="w-full rounded-md border border-gray-300 bg-white px-3 py-2 text-sm text-gray-900 placeholder:text-gray-400 focus:border-brand-blue focus:outline-none focus:ring-1 focus:ring-brand-blue dark:border-zinc-700 dark:bg-zinc-800 dark:text-zinc-100"
+            />
+            <p id="donor-email-hint" className="text-xs text-gray-500 dark:text-gray-400">
+              Required for payment receipt and donation tracking.
+            </p>
+          </div>
+        )}
```
[the remark this answers is not in the document]
Comment (brunodmsi) [reply components/Donation/SingleProject/OnrampFlow.tsx]: Already resolved. The email input now includes `aria-required="true"`, `aria-invalid`, `aria-describedby` pointing to either the error or hint element, and a visible error message with `role="alert"` — see `OnrampFlow.tsx:205-214` at current HEAD.

File: hooks/donation/types.ts [hunk 2]
```diff
@@ -62,10 +63,20 @@ export interface OnrampSessionResponse {
...
 }
 
+export interface DonationStatusApiResponse {
+  status: "pending" | "completed" | "failed";
+  amount?: string;
+  tokenSymbol?: string;
+  transactionHash?: string;
+  fiatAmount?: number;
+  fiatCurrency?: string;
+}
```
[the remark this answers is not in the document]
Comment (brunodmsi) [reply hooks/donation/types.ts]: Already resolved. `DonationStatusApiResponse.status` is now typed as `DonationStatus` (the enum) directly, not string literals — see `types.ts:72` at current HEAD.

File: hooks/donation/__tests__/useDonationPolling.test.tsx [hunk 3]
```diff
@@ -209,4 +209,123 @@ describe("useDonationPolling", () => {
...
+      );
+
+      await waitFor(() => {
+        expect(result.current.donation).not.toBeNull();
+      });
+
+      const callCount = mockDonationsService.getDonationStatus.mock.calls.length;
+      await new Promise((r) => setTimeout(r, 100));
+      expect(mockDonationsService.getDonationStatus.mock.calls.length).toBe(callCount);
+    });
```
[the remark this answers is not in the document]
Comment (brunodmsi) [reply hooks/donation/__tests__/useDonationPolling.test.tsx]: Acknowledged. This is about the test file using `setTimeout` under fake timers — will look into adjusting the test to use `jest.advanceTimersByTime` instead. The other items flagged in this review (type safety, a11y, etc.) are all already resolved.

**Review by amaurymagalhaes — COMMENTED** [review]
## PR Review: `feat(donations): add unauthenticated fiat donation flow`

**+375 / -55 | 14 files changed**

### Summary

Solid, well-scoped feature adding unauthenticated fiat donations via email + Stripe. The code is clean, well-tested, and follows existing patterns. The PR description is excellent — one of the most thorough I've seen. A few issues worth addressing before merge, but nothing blocking.

---

### Critical Issues (P0/P1)

**1. [P1] Email regex allows invalid edge cases**

`OnrampFlow.tsx`:
```js
if (!/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(donorEmail))
```

This regex allows strings like `user@.com`, `user@-domain.com`, `user@domain..com`, and `user..name@domain.com`. While no regex perfectly validates email, this one is more permissive than it looks. **Is the backend also validating before creating the Stripe session?** If yes, this is fine as a UX hint (downgraded to P3). If not, malformed emails reaching Stripe will fail silently or cause receipt failures.

---

**2. [P1] `pollingToken` included in React Query cache key — contradicts stated security goal**

`useDonationPolling.ts`:
```ts
queryKey: pollingToken
  ? QUERY_KEYS.DONATIONS.STATUS(donationUid!, chainId, pollingToken)
  : QUERY_KEYS.DONATIONS.POLLING(donationUid!, chainId),
```

`queryKeys.ts`:
```ts
STATUS: (uid: string, chainId: number, token: string) =>
  ["donation-status", uid, chainId, token] as const,
```

The `pollingToken` is stored in the React Query cache key, making it visible in React Query DevTools and any debugging/monitoring tools. The PR description specifically calls out avoiding query params to prevent token leaking — yet the token is now in the client-side cache key which is equally visible.

**Fix**: Remove the token from the cache key. There should only be one active polling status per donation, so the key doesn't need the token for uniqueness:

```ts
STATUS: (uid: string, chainId: number) =>
  ["donation-status", uid, chainId] as const,
```

The token is already passed to the `queryFn` — it doesn't need to be in the key for correctness.

---

### Other Issues (P2/P3)

**3. [P2] `publicApiClient` error interceptor uses `console.error` — ships to production**

`donations.service.ts`:
```ts
(error) => {
  console.error("Donation Status API Error:", error.response?.data || error.message);
  throw error;
}
```

This follows the existing pattern (authenticated `apiClient` has the same interceptor), so it's consistent. But both will appear in production browser consoles. Consider using a logging service or guarding with `process.env.NODE_ENV !== 'production'`.

**4. [P3] Unrelated changes bundled in PR**

- `.claude/settings.local.json`: Formatting-only change — shouldn't be in a feature PR
- `__tests__/services/programRegistry.service.test.ts`: Removal of `anyoneCanJoin: false` — unrelated test fix

These should ideally be in separate commits to keep the diff focused.

**5. [P3] `pollingToken ?? undefined` conversion is slightly awkward**

`OnrampSuccessModal.tsx`:
```ts
pollingToken: pollingToken ?? undefined,
```

The prop type is `pollingToken?: string | null` but the hook param is `pollingToken?: string`. Consider making the hook param accept `null` too for consistency, or keep as-is (functional either way).

---

### What's Good

- `toDonationStatus()` type guard replacing unsafe `as DonationStatus` cast — correct pattern
- `X-Polling-Token` header instead of query params — good security practice
- `emailTouched` with `onBlur` — avoids premature validation UX
- Comprehensive test coverage for both authenticated and unauthenticated paths
- `encodeURIComponent(uid)` in `getDonationStatus` — good defensive coding
- PR description is exceptionally thorough

### Suggestions

- Add a test for email trimming behavior (e.g., `" test@example.com "` → button enabled)
- Confirm backend validates email format before Stripe session creation

---

### Overall Score: 8/10

### Merge Recommendation: **Approve with minor changes**

Address #2 (token in cache key) before merge — it contradicts the stated security goal. #1 is fine if backend validates. Everything else is minor polish.
